internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go

```diff
@@ -653,7 +653,7 @@ func TestServerSideApplyWithDefaulting(t *testing.T) {
 
 			// Enable the new defaulting logic (i.e. simulate the Cluster API update).
 			// The webhook will default the users field to `[{Name: "default-user"}]`.
-			g.Expect(env.Create(ctx, mutatingWebhookConfiguration)).To(Succeed())
+			g.Expect(env.CreateAndWait(ctx, mutatingWebhookConfiguration)).To(Succeed())
```

Comment:
Sorry but this doesn't make sense.

The test is failing in l.683 because the KubeadmConfigTemplate doesn't exist in the local cache. Here we are waiting for MutatingWebhookConfiguration.

It probably fixes the issue by accident but I would prefer waiting for the right object (which is kct)

Reply:
Let's add the following instead in l.652
```go
			// Let's wait until the initial KubeadmConfigTemplate is visible in the local cache. Otherwise the test fails below.
			g.Eventually(ctx, func(g Gomega) {
				g.Expect(env.Get(ctx, client.ObjectKeyFromObject(kct), &bootstrapv1.KubeadmConfigTemplate{})).To(Succeed())
			}, 5*time.Second).Should(Succeed())
```

Reply:
I have added the wait for kct too.

Reply:
Let's get rid of the CreateAndWait in l.661. We absolutely don't depend on mutatingWebhookConfiguration showing up in the local client cache

Reply:
done